src/proxy/inbound.rs

```diff
@@ -53,17 +52,16 @@ impl Inbound {
         Ok(Inbound {
             cfg,
             workloads,
-            listener,
             cert_manager,
         })
     }
 
     pub(super) fn address(&self) -> SocketAddr {
-        self.listener.local_addr().unwrap()
+        self.cfg.inbound_addr
```

Comment:
This is not correct, this is the configured address -- if we bind to port `0` it will not return the real port

Reply:
If it is 0， then how does the redirect configurred

Reply:
Bind to 0 is for tests. We already do this but there was a small issue in the test so I had to skip it. It would have detected this and failed if I hadn't skipped though.

Reply:
IC

Reply:
Hi @howardjohn, okay, got it. So the OS need to assigns a port to Inbound when binding to port `0`. I will check it here.

Reply:
We also need to bind outside of run() not just check something here

Reply:
Ok, so that's why we need to keep a tcp listener in Inbound, am I correct?

Reply:
Yes